bokeh/events.py

```diff
@@ -7,90 +7,130 @@
 callbacks can be attached and executed when they occur.
 
 It is possible to respond to events with ``CustomJS`` callbacks, which will
-function with or without a Bokeh server. This can be accomplished with the
-:func:`~bokeh.model.Model.js_on_event` method:
+function with or without a Bokeh server. This can be accomplished by passing
+and event class, and a ``CustomJS`` model to the
+:func:`~bokeh.model.Model.js_on_event` method. When the ``CustomJS`` is
+executed in the browser, its ``cb_obj`` argument will contain the concrete
+event object that triggered the callback.
 
 .. code-block:: python
 
-    from bokh.models import Button
+    from bokeh.events import ButtonClick
+    from bokeh.models import Button, CustomJS
+
     button = Button()
-    button.js_on_event(events.ButtonClick, CustomJS(code='console.log("JS:Click")'))
+
+    button.js_on_event(ButtonClick, CustomJS(code='console.log("JS:Click")'))
 
 Alternatively it is possible to trigger Python code to run when events
 happen, in the context of a Bokeh application running on a Bokeh server.
-This can ccomplished with the :func:`~bokeh.model.Model.on_event` method:
+This can accomplished by passing an event class, and a callback function
+to the the :func:`~bokeh.model.Model.on_event` method. The callback should
+accept a single argument ``event``, which will be passed the concrete
+event object that triggered the callback.
+
+.. code-block:: python
 
-. code-block:: python
+    from bokeh.events import ButtonClick
+    from bokeh.models import Button
 
-    from bokh.models import Button
     button = Button()
 
-    def click_callback(event):
-       print('Python:Click')
+    def callback(event):
+        print('Python:Click')
 
-    button.on_event(events.ButtonClick, click_callback)
+    button.on_event(ButtonClick, callback)
+
+.. note ::
+    There is no throttling of events. Some events such as ``MouseMove``
+    may trigger at a very high rate.
 
 '''
 from __future__ import absolute_import
-from .util.future import with_metaclass
+
 import logging
 logger = logging.getLogger(__file__)
 
-EVENT_CLASSES = set()
-
+from .util.future import with_metaclass
 
+_CONCRETE_EVENT_CLASSES = dict()
```

Comment:
I also changed this, in two ways: a dict instead of a set (makes looking up event names cleaner) and also it only stores concrete event types ("abstract" event classes with no event_name) do not get put in the index.